pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarStreamMetadataProvider.java

```diff
@@ -156,11 +174,21 @@ public List<PartitionGroupMetadata> computePartitionGroupMetadata(String clientI
       LOGGER.warn("Error encountered while calculating pulsar partition group metadata: " + e.getMessage(), e);
     } finally {
       closeConsumer(consumer);
+      deleteSubscription(_topic, subscription);
     }
 
     return newPartitionGroupMetadataList;
   }
 
+  private void deleteSubscription(String topicName, String subscription) {
+    try {
+      _pulsarAdminClient.topics().deleteSubscription(topicName, subscription);
```

Comment:
I saw that it was possible to create lightweight subscriptions using non durable subscription mode, it may be a simpler / more efficient solution to solve the problem (the subscription is never persisted):

        Consumer<byte[]> consumer = pulsarClient.newConsumer()
                .topic("my-topic")
                .subscriptionName("my-sub")
                .subscriptionMode(SubscriptionMode.NonDurable)
                .subscribe();

Reply:
> I think we should create them non-durable, but maybe also delete them to clean the memory of the Pulsar brokers. 

I wasn't aware of this mode. But what is the benefit of creating them non-durable, if we have to delete them anyway?

Reply:
@navina non durable subscriptions are not persisted, so I suppose there are no cursors written on BookKeeper. I assume, without having checked, that this makes it possible to create and delete subscriptions more quickly.

Reply:
@navina non durable subscriptions are not persisted, so I suppose there are no cursors written on BookKeeper. I assume, without having checked, that this makes it possible to create and delete subscriptions more quickly.

Reply:
I couldn't find any doc around this and haven't got much response in pulsar slack. since I have to look at the pulsar code next, how about we re-visit this and leave the implementation in the PR as it is, @mathieudruart ?

Reply:
@navina yes I suppose we can

Reply:
IMO, we can make this change. This subscription is only used to fetch metadata so it is fine if its non durable.
https://pulsar.apache.org/docs/2.10.x/concepts-messaging/#subscription-modes

Reply:
Other than that, PR LGTM!

Reply:
I haven't tested, but maybe a non-durable subscription doesn't need to be deleted? maybe it is automatically deleted once consumer is closed?

Reply:
>I haven't tested, but maybe a non-durable subscription doesn't need to be deleted? maybe it is automatically deleted once consumer is closed?

Ok. I will test it today as I have some time :)

Reply:
Alright. I was able to verify with pulsar standalone mode that the non-durable subscription gets instantly deleted if the consumer is closed. 👏 

@mathieudruart the actual root cause of all those unused subscriptions not getting cleaned up on the broker is due to the fact that we are not closing the consumer properly in the metadata provider. I have already fixed that in the PR. 

Will update the PR to use NonDurable subscription mode and remove the explicit delete.